packages/blue-sdk-wagmi/src/hooks/useChainId.ts

```diff
@@ -0,0 +1,19 @@
+import { ChainUtils } from "@morpho-org/blue-sdk";
+import { ChainIdParameter } from "@wagmi/core/internal";
+import {
+  Config,
+  ResolvedRegister,
+  useChainId as wagmi_useChainId,
+} from "wagmi";
+import { ConfigParameter } from "../types/properties.js";
+
+export type UseChainIdParameters<config extends Config = Config> =
+  ChainIdParameter<config> & ConfigParameter<config>;
+
+export function useChainId<config extends Config = ResolvedRegister["config"]>(
+  parameters: UseChainIdParameters<config>,
+) {
+  const wagmiChainId = wagmi_useChainId(parameters);
+
+  return ChainUtils.parseSupportedChainId(parameters.chainId ?? wagmiChainId);
```

Comment:
Is it convenient to have react hooks throw errors?

Reply:
Nice catch, I'll try to find a workaround

Reply:
I actually think it's fine to break everything if wagmi is connected to a chain id not accepted by the sdk